### PR TITLE
Updating Daml VS Code plugin from upstream Haskell upstream along with constructs in DAML (choice, key) 

### PR DESCRIPTION
**Pull request by akalburgi**
Updating Daml VS Code plugin from upstream Haskell upstream along with constructs in DAML (choice, key) 

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

### CODE REVIEW

**Comment by akalburgi** [comment]
> Why rename the file `-update` at the same time? The changes don't look unreasonable

the plan is to delete the -update one after I get a working one out, Wanted to have the old one for reference. Will delete it before merging

**Comment by neil-da** [comment]
Makes sense!

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml [hunk 1]
```diff
@@ -1,89 +1,84 @@
-<!-- Copyright (c) 2017 Justus Adam -->
-
-<!-- All rights reserved. -->
-
-<!-- Redistribution and use in source and binary forms, with or without -->
-<!-- modification, are permitted provided that the following conditions are met: -->
-
-<!--     * Redistributions of source code must retain the above copyright -->
-<!--       notice, this list of conditions and the following disclaimer. -->
```
Comment (martin-drhu-da) [inline daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml]: Are violating this rule by removing the copy right header?

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml [hunk 2]
```diff
@@ -463,7 +456,7 @@
                   )
                   (?:\s*,\s*\g&lt;fn&gt;)?
                 )
-                \s*(::|∷)</string>
+                \s*(:|:)</string>
```
Comment (martin-drhu-da) [inline daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml]: Isn't `:|:` equivalent to just `:`?

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml [hunk 3]
```diff
@@ -514,14 +511,22 @@
 			<string>#infix_op</string>
 		</dict>
 		<dict>
-			<key>begin</key>
+			<key>comment</key>
+			<string>We treat :: as cons. It's important this definition precede the rule for : (and that rule precede the one for arbitrary operators).</string>
+			<key>match</key>
 			<string>(::|∷)</string>
```
Comment (martin-drhu-da) [inline daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml]: I'm not sure we really want the unicode version of `::`. @neil-da @shayne-fletcher-da any thoughts?

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml (hunk 1 above)
Comment (neil-da) [reply daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml]: Yep, let's leave this

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml (hunk 3 above)
Comment (neil-da) [reply daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml]: Yeah, let's ban Unicode other than in identifiers (and even then, I'd rather than it universally...). However, I think the grammar probably permits it, so might as well have the syntax match the grammar, and removing it from the grammar seems like the wrong call without persuading GHC upstream to drop silly unicode.

TLDR: I'd keep it.

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml (hunk 2 above)
Comment (akalburgi) [reply daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml]: Ah yes, thanks for pointing out, will fix it

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml (hunk 1 above)
Comment (akalburgi) [reply daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml]: this is been taken off from the haskell vs code plugin grammar file https://github.com/JustusAdam/language-haskell/blob/master/syntaxes/haskell.tmLanguage

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml (hunk 3 above)
Comment (shayne-fletcher-da) [reply daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml]: Its straight from Justus's work. Let's us not remove it here.

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml [hunk 4]
```diff
@@ -514,14 +511,22 @@
...
+		<dict>
+			<key>begin</key>
+			<string>(:)</string>
 			<key>beginCaptures</key>
 			<dict>
 				<key>1</key>
 				<dict>
 					<key>name</key>
-					<string>keyword.other.double-colon.haskell</string>
+					<string>keyword.other.double-colon.daml</string>
```
Comment (shayne-fletcher-da) [inline daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml]: @akalburgi as we discussed, are we not going to change this (similarly line 481 at least)?

File: daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml (hunk 3 above)
Comment (shayne-fletcher-da) [reply daml-foundations/daml-tools/daml-extension/syntaxes/daml12.tmLanguage-update.xml]: FWIW, @neil-da recent changes in the GHC parser suggest that the community is in fact "doubling-down" on the unicode silliness :(

**Comment by akalburgi** [comment]
fixes #343
